### PR TITLE
[FE-641] moved filter resources via owner ids to the common module

### PR DESCRIPTION
**Pull request by anandjakhaniya**
## Impact and Context
Proto PR: https://github.com/VertaAI/protos-all/pull/430
UAC PR: https://github.com/VertaAI/verta-uacservice/pull/2041

**Description:**
- Moved https://github.com/VertaAI/modeldb/pull/3529/files#diff-ea7810799fd436074c6a2ebe97e08246cc8f1ff57293f9fd6e123467cb453faeL260 function to the common module and refactored that function to use it as common code.
- refactor test cases to check the modified first point.

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

File: backend/common/src/main/java/ai/verta/modeldb/common/authservice/UACApisUtil.java [hunk 1]
```diff
@@ -42,10 +47,12 @@ public class UACApisUtil {
   private static final Logger LOGGER = LogManager.getLogger(UACApisUtil.class);
   protected final FutureExecutor executor;
   protected final UAC uac;
+  protected final boolean isPermissionV2;
```
Comment (evgeniilezhnin) [inline backend/common/src/main/java/ai/verta/modeldb/common/authservice/UACApisUtil.java]: Why is this flag needed? I thought search by owner works for both permission v1 and v2?

**Review by evgeniilezhnin — APPROVED** [review]

File: backend/common/src/main/java/ai/verta/modeldb/common/authservice/UACApisUtil.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/common/src/main/java/ai/verta/modeldb/common/authservice/UACApisUtil.java]: It is not in used so removed

**Review by evgeniilezhnin — APPROVED** [review]

**Comment by anandjakhaniya** [comment]
@eheinlein-verta

**Review by eheinlein-verta — APPROVED** [review]
I'll approve this so it is not a blocker for anyone. But could you do the following:

- Add some more context and testing details to the overview of the PR
- Add some comments (in the PR, not the code) explaining what code was simply moved vs. what code was actually changed

**Pull request merged**